docs/src/main/asciidoc/cache.adoc

```diff
@@ -366,6 +378,106 @@ public class CachedService {
 <3> Calling this method WILL invalidate values cached by the `load` method because the key elements order is the same.
 <4> Calling this method WILL NOT invalidate values cached by the `load` method because the key elements order is different.
 
+=== Generating a cache key with `CacheKeyGenerator`
+
+You may want to include more than the arguments of a method into a cache key.
+This can be done by implementing the `io.quarkus.cache.CacheKeyGenerator` interface and declaring that implementation in the `keyGenerator` field of a `@CacheResult` or `@CacheInvalidate` annotation.
+
+If a CDI scope is declared on a key generator, then it will be injected as a CDI bean during the cache key computation.
```

Comment:
So I suppose that only class-based [managed beans](https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#managed_beans) are supported, right? I.e. no producers or synthetic beans.

Also it seems that you don't take CDI qualifiers into account at runtime and so you should probably forbid the usage of custom qualifiers, or just mention that the _default qualifier_ must be used (in CDI this means either no qualifier annotation or `@javax.enterprise.inject.Default`).

Reply:
Yes, the key generators are restricted to class-based managed beans because we need to declare the keygen class in the caching annotations.

Qualifiers should be forbidden indeed and/or documented.